src/newViews/Tagging/Sources/index.tsx

```diff
@@ -31,10 +31,17 @@ function Sources(props: Props) {
         hideSingleSourceAddModal,
     ] = useModalState(false);
 
+    const [
+        isBulkModalShown,
+        showBulkUploadModal,
+        hideBulkUploadModal,
+    ] = useModalState(false);
+
     return (
         <div className={styles.sources}>
             <Navbar
                 onAddSingleSourceClick={showSingleSourceAddModal}
+                onBulkUploadClick={showBulkUploadModal}
             />
```

Comment:
Just a question. Why is Navbar inside one of the tabs? Shouldn't it be where the tabs are used? In Tagging/index.tsx perhaps?

Reply:
@subinasr it's because the navbar must be rendered in every tab of Tagging page. Furthermore, each tab must have a permanent route. Under these constraints, this is a simple way. If we are to handle nested navigation we need to refactor a lot of the main Navbar component.

Reply:
Thanks man, that helped.